client/src/js/components/bhTransactionTypeSelect.js

```diff
@@ -0,0 +1,58 @@
+angular.module('bhima.components')
+  .component('bhTransactionTypeSelect', {
+    templateUrl : 'modules/templates/bhTransactionTypeSelect.tmpl.html',
+    controller  : transactionTypeSelectController,
+    bindings    : { 
+      transactionTypes : '<?',
+      label            : '@?',
+      onSelectCallback : '&',
```

Comment:
Would it not be better to just have a `onUpdate()` or `onChanges()` binding that simply returns the list of selected transaction types?